FormidableLabs/spectacle
# Feature/autoplay toggle on start

### Description

1. Alt/Option + A Start autoplay should be a toggle (start/stop depending on the state - same as the other key combination toggles)

2. Add prop/tag to Deck to pause autoplaying on slideshow start

Fixes https://github.com/FormidableLabs/spectacle/issues/633 (issue)

#### Type of Change

Please delete options that are not relevant.


### How Has This Been Tested?

Master branch has been downloaded and functionality has been tested by manually trying the 2 new features in the example project. To verify changes:

1. Enable autoplay in the example presentation and start/pause autoplay multiple times with key combination Alt/Option + A

2. Setup Deck props with all combinations of autoplay={true/false} and autoplayOnStart={true/false}
- Case autoplay={true} autoplayOnStart={false} -> presentation will be unpaused on start (can be paused with key combination & click)
- Case autoplay={true} autoplayOnStart={true} -> presentation will be paused on start (can be unpaused with key combination & click)
-Case autoplay={false} autoplayOnStart={false} -> autoplay controls are not available & no autoplay
-Case autoplay={false} autoplayOnStart={true} -> autoplay controls are not available & no autoplay (autoplayOnStart is ignored)

### Checklist:

- [x] My code follows the style guidelines of this project (I have run `yarn prettier-fix && yarn lint`)
- [x] New and existing unit tests pass locally with my changes (I have run `yarn test`)
